pkg/matcher/cel.go

```diff
@@ -24,30 +24,53 @@ const (
 	changedFilesTags = "files."
 )
 
+// getBranchFromExpression function extracts the value associated with the provided branchKey.
+//
+// ex: Given an expression like target_branch == "main", getBranchFromExpression returns "main".
+func getBranchFromExpression(branchKey string) string {
+	parts := strings.Split(branchKey, "==")
+
+	var branch string
+	if len(parts) == 2 {
+		// Trim any leading or trailing white spaces from the value
+		value := strings.TrimSpace(parts[1])
+		sVal := strings.Split(value, `"`)
+		// Only valid values for example, are target_branch == "main" or source_branch == "*" etc..
+		// Invalid formats include target_branch == ["main"] or target_branch == main.
+		// For valid values, the length is always 3 when splitting "main" by `"` character.
+		if len(sVal) == 3 {
+			branch = sVal[1]
+		}
+	}
+	return branch
+}
+
 func celEvaluate(ctx context.Context, expr string, event *info.Event, vcx provider.Interface) (ref.Val, error) {
 	eventTitle := event.PullRequestTitle
-	if event.TriggerTarget == triggertype.Push {
-		eventTitle = event.SHATitle
-		// For push event the target_branch & source_branch info coming from payload have refs/heads/
-		// but user may or mayn't provide refs/heads/ info while giving target_branch or source_branch in CEL expression
-		// ex:  pipelinesascode.tekton.dev/on-cel-expression: |
-		//        event == "push" && target_branch == "main" && "frontend/***".pathChanged()
-		// This logic will handle such case.
-		splittedValue := strings.Split(expr, "&&")
-		for i := range splittedValue {
-			if strings.Contains(splittedValue[i], "target_branch") {
-				if !strings.Contains(splittedValue[i], "refs/heads/") {
-					event.BaseBranch = strings.TrimPrefix(event.BaseBranch, "refs/heads/")
-				}
+	var (
+		targetBranchValue string
+		sourceBranchValue string
+	)
+	splitValue := strings.Split(expr, "&&")
+	for i := range splitValue {
+		if strings.Contains(splitValue[i], "target_branch") {
```

Comment:
better to use `HasPrefix`

Reply:
@piyush-garg I reverted back to use strings.Contains because
strings.HasPrefix works if target_branch is the prefix
But we have a scenario like below
```
event == "push" && ( target_branch == "main" ) && "components/?*/base/**".pathChanged()
```
where HasPrefix won't work